### PR TITLE
Clean up MetricCollection

### PR DESCRIPTION
**Pull request by udoprog**
This converts up the current `abstract` MetricCollection into an interface, and uses a standard serialization method in fasterxml instead of the custom serializer (see removal of `MetricCollectionSerialization`).

This is in preparation of experimenting with additional Metric types, in which modifying MetricCollection would be a central change. Simplifying and streamlining this will improve the clarity of future pull requests.

While doing this, a few bugs to existing Metrics were detected, and tests introduced in `BasicSerialization`.

The `EmptyMetricCollection` null object has been removed. Attempting to share it between nodes results in inconsistent behavior, which is just confusing.

### CODE REVIEW

File: heroic-component/src/test/java/com/spotify/heroic/aggregation/AggregationOutputTest.java [hunk 1]
```diff
@@ -20,16 +20,16 @@ public void lombokDataTest() {
 
     @Test
     public void isEmpty() {
-        final AggregationOutput output =
-            new AggregationOutput(ImmutableMap.of(), ImmutableSet.of(), MetricCollection.empty());
+        final AggregationOutput output = new AggregationOutput(ImmutableMap.of(), ImmutableSet.of(),
+            MetricCollection.points(ImmutableList.of()));
```
Comment (gabrielgerhardsson) [inline heroic-component/src/test/java/com/spotify/heroic/aggregation/AggregationOutputTest.java]: Say, what's the reason behind removing the EmptyMetricCollection class?

File: heroic-core/src/main/java/com/spotify/heroic/http/write/WriteResource.java [hunk 2]
```diff
@@ -52,7 +60,13 @@ public void metrics(
         @Suspended final AsyncResponse response, @QueryParam("group") String group,
         WriteMetricRequest write
     ) throws Exception {
-        httpAsync.bind(response, ingestion.useGroup(group).write(write.toIngestionRequest()),
-            r -> r);
+        final IngestionGroup ingestionGroup = ingestion.useGroup(group);
```
Comment (gabrielgerhardsson) [inline heroic-core/src/main/java/com/spotify/heroic/http/write/WriteResource.java]: Unrelated change to this commit?

File: heroic-component/src/test/java/com/spotify/heroic/aggregation/AggregationOutputTest.java (hunk 1 above)
Comment (udoprog) [reply heroic-component/src/test/java/com/spotify/heroic/aggregation/AggregationOutputTest.java]: Unnecessary null object. It's also mentioned in the description:

> The EmptyMetricCollection null object has been removed. Attempting to share it between nodes results in inconsistent behavior, which is just confusing.

File: heroic-core/src/main/java/com/spotify/heroic/http/write/WriteResource.java (hunk 2 above)
Comment (udoprog) [reply heroic-core/src/main/java/com/spotify/heroic/http/write/WriteResource.java]: This happens because `EmptyMetricCollection` was removed in favor of `Optional<MetricCollection>`

File: heroic-component/src/test/java/com/spotify/heroic/aggregation/AggregationOutputTest.java (hunk 1 above)
Comment (gabrielgerhardsson) [reply heroic-component/src/test/java/com/spotify/heroic/aggregation/AggregationOutputTest.java]: Ok I see, thanks. Sorry, didn't see the description note regarding that.

File: heroic-core/src/main/java/com/spotify/heroic/http/write/WriteResource.java (hunk 2 above)
Comment (gabrielgerhardsson) [reply heroic-core/src/main/java/com/spotify/heroic/http/write/WriteResource.java]: Ok 👍
